frontend/src/container/IngestionSettings/MultiIngestionSettings.tsx

```diff
@@ -0,0 +1,1133 @@
+						Update Ingestion Key
+					</Button>,
+				]}
+			>
+				<Form
+					name={`edit-ingestion-key-form-${activeAPIKey?.id}`}
+					key={activeAPIKey?.id}
+					form={editForm}
+					layout="vertical"
+					autoComplete="off"
+					initialValues={{
+						name: activeAPIKey?.name || '',
+						tags: activeAPIKey?.tags || [],
+						expires_at: dayjs(activeAPIKey?.expires_at) || null,
+					}}
+				>
+					<Form.Item
+						name="name"
+						label="Name"
+						rules={[{ required: true }, { type: 'string', min: 6 }]}
+					>
+						<Input placeholder="Enter Ingestion Key name" disabled />
+					</Form.Item>
+
+					<Form.Item name="tags" label="Tags">
+						<Tags tags={updatedTags} setTags={setUpdatedTags} />
+					</Form.Item>
+
+					<Form.Item
+						className="expires-at"
+						name="expires_at"
+						label="Expiration"
+						rules={[{ required: true }]}
+					>
+						<DatePicker
+							popupClassName="ingestion-key-expires-at"
+							disabledDate={disabledDate}
+						/>
+					</Form.Item>
+				</Form>
+			</Modal>
+
+			{/* Create New Key Modal */}
+			<Modal
+				className="ingestion-key-modal"
+				title="Create new ingestion key"
+				open={isAddModalOpen}
+				key="create-ingestion-key-modal"
+				closable
+				onCancel={hideAddViewModal}
+				destroyOnClose
+				footer={[
+					<Button
+						key="cancel"
+						onClick={hideAddViewModal}
+						className="periscope-btn cancel-btn"
+						icon={<X size={16} />}
+					>
+						Cancel
+					</Button>,
+					<Button
+						className="periscope-btn primary"
+						test-id="create-new-key"
+						key="submit"
+						type="primary"
+						icon={<Check size={14} />}
+						loading={isLoadingCreateAPIKey}
+						onClick={onCreateIngestionKey}
+					>
+						Create new Ingestion key
+					</Button>,
+				]}
+			>
+				<Form
+					key="createForm"
+					name="create-ingestion-key-form"
+					form={createForm}
+					initialValues={{
+						role: USER_ROLES.ADMIN,
+						expiration: '1',
+						name: '',
+					}}
+					layout="vertical"
+					autoComplete="off"
+				>
+					<Form.Item
+						name="name"
+						label="Name"
+						rules={[
+							{ required: true },
+							{ type: 'string', min: 6 },
+							{
+								pattern: /^[a-zA-Z0-9_-]*$/,
+								message:
+									'Ingestion key name should only contain letters, numbers, underscores, and hyphens.',
+							},
+						]}
+						validateTrigger="onBlur"
+					>
+						<Input placeholder="Enter Ingestion Key name" autoFocus />
+					</Form.Item>
+
+					<Form.Item
+						className="expires-at"
+						name="expires_at"
+						label="Expiration"
+						rules={[{ required: true }]}
+					>
+						<DatePicker
+							popupClassName="ingestion-key-expires-at"
+							disabledDate={disabledDate}
+						/>
+					</Form.Item>
+
+					<Form.Item name="tags" label="Tags">
+						<Tags tags={updatedTags} setTags={setUpdatedTags} />
+					</Form.Item>
+				</Form>
+			</Modal>
+		</div>
+	);
+}
+
+export default MultiIngestionSettings;
```

Comment:
NIT: We can divide this file into multiple, especially around the render return we have too many nested divs, so we can move those to multiple components if possible for better readability.

Reply:
Yes. Will refactor this later